src/bin/scheduler/scheduling/scheduler.rs

```diff
@@ -80,7 +93,13 @@ async fn run_cleanup_job(cancellation_token: CancellationToken, plans: Vec<Plan>
             _ = clock.tick() => { }
             _ = cancellation_token.cancelled() => { return }
         };
-        spawn_blocking(move || cleanup_working_directories(plans.iter()));
```

Comment:
Wait, why does this not have an await? Did our cleanup job ever work?

Reply:
Yes, they did. You don't have to explicitly await this thread, it runs anyway. Adding the await avoids spawning multiple cleanup threads, but hopefully, this never happened anyway (they shouldn't take 5 minutes).